mudslide/models.py

```diff
@@ -553,12 +562,205 @@ def dV(self, X: ArrayLike) -> ArrayLike:
         return out.reshape([5, 2, 2])
 
 
+class TMModel(AdiabaticModel_):
+    def __init__(
+        self,
+        turbomole_dir: str,
+        states: ArrayLike,
+        sub_dir_stem:str = "traj", 
+        representation: str = "adiabatic",
+        reference: Any = None,
+    ):
+        AdiabaticModel_.__init__(self, representation=representation, reference=reference)
+
+        self.turbomole_dir = turbomole_dir
+        self.states = states
+        self.nstates_ = len(self.states)
+        self.sub_dir_stem = sub_dir_stem
+        self.sub_dir_num = 0
+
+        self.turbomole_init()
+
+    def turbomole_init(self):
+        self.coord_path = Path(self.turbomole_dir)/"control"  
+        self.get_coords()
+        self.call_and_parse_turbomole()
+
+    def get_coords(self):
+        with open(self.coord_path, "r") as coord_file:
+            lines = coord_file.readlines()
+
+        coords = []
+        self.atom_order = []
+
+        try:
+            read_start = np.argwhere(np.array(["$coord\n" in line for line in lines]))[0][0]
+        except:
+            
+            self.coord_path = Path(self.turbomole_dir)/"coord"
+            with open(self.coord_path, "r") as coord_file:
+                lines = coord_file.readlines()
+            read_start = np.argwhere(np.array(["$coord\n" in line for line in lines]))[0][0]
+
+        first_char = ""
+        current_line = read_start + 1
+
+        while True:
+            line = lines[current_line]
+            first_char = line[0]
+            if first_char == "$":
+                break
+            line_split = re.split("\s+", line.strip())
+            self.atom_order.append(line_split[-1])
+            coords.extend([float(val) for val in line_split[0:3]])
+            current_line += 1
+    
+        self.ndim_ = len(coords)
+        self.X = np.array(coords, dtype=np.float64).reshape(self.ndim())
+        self.mass = [masses[atom] for atom in self.atom_order]
+        self.mass = [3 * [masses[atom]] for atom in self.atom_order]
+        self.mass = np.array(self.mass, dtype=np.float64).reshape(self.ndim()) * amu_to_au
+
+
+    def update_coords(self, X):
```

Comment:
This is really weirdly hardcoded. Rewrite this using the built in formatting features in python. It will be like 4 lines long if you do that.

Reply:
Used python built in formatting!

Reply:
Use fixed widths and the scientific notation to make sure you don't lose any numerical precision. I think something like `{:20.16g} {:20.16g} {:20.16g} {:3s}` should work well.

Reply:
updated!

Reply:
The most recent version of the code still uses the `float` format (`{:22.14f}`). That will lead to precision loss for small values. Use the suggested format above, which will be guaranteed to preserve the full precision of the coordinates. If turbomole complains about the `e` in the output, just use `line.replace("e", "d")` to fix it.

Reply:
Use `16g`. To be on the safe side.

Reply:
Okay, after some testing, the better format string to use is `{: 22.16e}`.

Reply:
hmm that was not working and I think because of the space issue (no space between the coordinates) so I instead used line 112 
<img width="758" alt="Screen Shot 2022-04-19 at 2 04 38 PM" src="https://user-images.githubusercontent.com/46786940/164067520-904f658b-8ae9-4704-8136-fe626757bd35.png">

The coord now looks like this 

<img width="713" alt="Screen Shot 2022-04-19 at 2 05 19 PM" src="https://user-images.githubusercontent.com/46786940/164067613-e661b790-0708-4073-a089-49db39789faa.png">